tests/chains/Filecoin/AddressTests.cpp

```diff
@@ -64,37 +74,70 @@ static const std::string invalidAddresses[] = {
     "f15ihq5ibzwki2b4ep2f46avlkrqzhpqgtga7rdrr",
     "f24vg6ut43yw2h2jqydgbg2xq7x6f4kub3bg6as66",
     "f3vvmn62lofvhjd2ugzca6sof2j2ubwok6cj4xxbfzz4yuxfkgobpihhd2thlanmsh3w2ptld2gqkn2jvlss44",
+    "f410f2oekwcmo2pueydmaq53eic2i62crtbeyuzx2gma",
 };
 
 TEST(FilecoinAddress, IsValid) {
     for (const auto& test : validAddresses) {
-        ASSERT_TRUE(Address::isValid(test.encoded))
-            << "is_valid() != true: " << test.encoded << std::endl;
-        ASSERT_TRUE(Address::isValid(parse_hex(test.hex)))
-            << "is_valid() != true: " << test.hex << std::endl;
+        ASSERT_TRUE(Address::isValid(test.string))
+            << "isValid(string) != true: " << test.string;
+
+        const Data encodedBytes = parse_hex(test.encoded);
+        ASSERT_TRUE(Address::isValid(encodedBytes))
+            << "isValid(Data) != true: " << test.encoded;
+    }
+}
+
+TEST(FilecoinAddress, IsInvalid) {
+    for (const auto& address : invalidAddresses) {
+        ASSERT_FALSE(Address::isValid(address))
+            << "isValid(string) != false: " << address;
     }
-    for (const auto& address : invalidAddresses)
-        ASSERT_FALSE(Address::isValid(address)) << "is_valid() != false: " << address << std::endl;
 
     ASSERT_FALSE(Address::isValid(parse_hex("00"))) << "Empty varuint";
     ASSERT_FALSE(Address::isValid(parse_hex("00ff"))) << "Short varuint";
     ASSERT_FALSE(Address::isValid(parse_hex("00ff00ff"))) << "Varuint with hole";
     ASSERT_FALSE(Address::isValid(parse_hex("000101"))) << "Long varuint";
     ASSERT_FALSE(Address::isValid(parse_hex("000000"))) << "Long varuint";
-    ASSERT_FALSE(Address::isValid(parse_hex("00ffffffffffffffffff02"))) << "Overflow";
+    ASSERT_FALSE(Address::isValid(parse_hex("00ffffffffffffffffff80"))) << "Overflow";
+}
+
+TEST(FilecoinAddress, Equal) {
+    for (const auto& test : validAddresses) {
+        const Data encodedBytes = parse_hex(test.encoded);
+        const Address lhs(test.string), rhs(encodedBytes);
+        ASSERT_EQ(lhs, rhs) << "Address(string) != Address(Data)";
+    }
+}
+
+TEST(FilecoinAddress, ExpectedProperties) {
+    for (const auto& test : validAddresses) {
```

Comment:
in general u can use `auto&&` as a sugar syntax, compiler will choose to make it const for u